lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1511,6 +1564,31 @@ protected function _validateAndCompleteAssociationMapping(array $mapping)
         // the sourceEntity.
         $mapping['sourceEntity'] = $this->name;
 
+        if (
```

Comment:
Reuse the factored out condition method described above.

Reply:
Done